homeassistant/components/influxdb/sensor.py

```diff
@@ -163,13 +217,25 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
                 entities.append(InfluxSensor(hass, influx, query))
             else:
                 _LOGGER.error(NO_BUCKET_ERROR, query[CONF_BUCKET])
-    else:
+
+    if CONF_QUERIES in config:
         for query in config[CONF_QUERIES]:
             if query[CONF_DB_NAME] in influx.data_repositories:
                 entities.append(InfluxSensor(hass, influx, query))
             else:
                 _LOGGER.error(NO_DATABASE_ERROR, query[CONF_DB_NAME])
 
+    if CONF_QUERIES_RAW in config:
+        for query in config[CONF_QUERIES_RAW]:
+            if query[CONF_API_VERSION] == DEFAULT_API_VERSION:
+                if query[CONF_DB_NAME] in influx.data_repositories:
+                    entities.append(InfluxSensor(hass, influx, query))
+                else:
+                    _LOGGER.error(NO_DATABASE_ERROR, query[CONF_DB_NAME])
+
+            if query[CONF_API_VERSION] == API_VERSION_2:
+                entities.append(InfluxSensor(hass, influx, query))
```

Comment:
```suggestion
            if query[CONF_BUCKET] in influx.data_repositories:
                entities.append(InfluxSensor(hass, influx, query))
            else:
                _LOGGER.error(NO_BUCKET_ERROR, query[CONF_BUCKET])
```
Same check as in the queries flux part above

Reply:
Your right, many thanks for your review!